Reply/app/src/main/java/com/example/reply/ui/ReplyListContent.kt

```diff
@@ -114,31 +139,108 @@ fun ReplyListOnlyContent(
+}
+
+@Composable
+fun ReplyEmailDetail(
+    email: Email,
+    isFullScreen: Boolean = true,
+    modifier: Modifier = Modifier.fillMaxSize(),
+    onBackPressed: () -> Unit
+) {
+    LazyColumn(modifier = modifier.background(MaterialTheme.colorScheme.inverseOnSurface)) {
+        item { EmailDetailAppBar(email, isFullScreen) {
+            onBackPressed.invoke()
+        } }
+        items(email.threads) { email ->
+            ReplyEmailThreadItem(email = email)
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun EmailDetailAppBar(
+    email: Email,
+    isFullScreen: Boolean,
+    onBackPressed: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(24.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        if (isFullScreen) {
+            FilledIconButton(
+                onClick = { onBackPressed.invoke() },
+                modifier = Modifier.padding(8.dp),
+                colors = IconButtonDefaults.filledIconButtonColors(
+                    containerColor = MaterialTheme.colorScheme.surface,
+                    contentColor = MaterialTheme.colorScheme.onSurface
+                )
+            ) {
+                Icon(
+                    imageVector = Icons.Default.ArrowBack,
+                    contentDescription = stringResource(id = R.string.back_button),
+                    modifier = Modifier.size(14.dp)
+                )
             }
         }
+        Column(
+            modifier = Modifier.weight(1f),
+            horizontalAlignment = if (isFullScreen) Alignment.CenterHorizontally else Alignment.Start
+        ) {
+            Text(text = email.subject, style = MaterialTheme.typography.titleMedium)
+            Text(
+                modifier = Modifier.padding(top = 4.dp),
+                text = "${email.threads.size} ${stringResource(id = R.string.messages)}",
+                style = MaterialTheme.typography.labelMedium,
+                color = MaterialTheme.colorScheme.outline
+            )
+        }
+        IconButton(
+            onClick = { /*TODO*/ },
+        ) {
+            Icon(imageVector = Icons.Default.MoreVert, contentDescription = stringResource(id = R.string.more_options_button))
+        }
     }
 }
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun ReplyEmailListItem(
     email: Email,
-    modifier: Modifier = Modifier.padding(horizontal = 16.dp, vertical = 4.dp)) {
-    Card(modifier = modifier) {
+    isSelected: Boolean = false,
+    modifier: Modifier = Modifier.padding(horizontal = 16.dp, vertical = 4.dp),
+    navigateToDetail: (Long) -> Unit
+) {
+    Card(
+        modifier = modifier.clickable { navigateToDetail.invoke(email.id) },
+        colors = CardDefaults.cardColors(
+            containerColor = if (isSelected) MaterialTheme.colorScheme.primaryContainer else MaterialTheme.colorScheme.surfaceVariant
```

Comment:
It'd be great if this selected also hooked into the semantics selection for accessibility.

One caveat there, is we would only want that to be displayed when both the list and detail are showing, since  "selected" would be a bit weird if only the list is showing.

Reply:
Yes it's only for detail only actually, you won't see for the single list.

Reply:
Added semantics for selected 👍🏽

Reply:
Could we only add the semantics if we're showing both the list and detail? Right now, the semantics is added for both, so on the list-only screen, we're getting the `Not selected`/`selected` read out.

Reply:
done 👍🏽